src/Raven.Studio/typescript/widgets/virtualGrid/columns/multiNodeTagsColumn.ts

```diff
@@ -0,0 +1,65 @@
+﻿/// <reference path="../../../../typings/tsd.d.ts"/>
+
+import virtualGridController = require("widgets/virtualGrid/virtualGridController");
+import virtualColumn = require("widgets/virtualGrid/columns/virtualColumn");
+import taskItem = require("models/resources/widgets/taskItem");
+import generalUtils = require("common/generalUtils");
+
+class multiNodeTagsColumn implements virtualColumn {
+
+    constructor(gridController: virtualGridController<taskItem>,
+                public width: string,
+                public opts: textColumnOpts<taskItem> = {}) {
+    }
+
+    canHandle(actionId: string): boolean {
+        return false;
+    }
+
+    get sortable() {
+        return false;
+    }
+
+    header = `<div>Nodes</div>`;
+
+    get headerAsText() {
+        return this.header;
+    }
+
+    get headerTitle() {
+        const titleToUse = this.opts && this.opts.headerTitle ? this.opts.headerTitle : this.header;
+        return generalUtils.unescapeHtml(titleToUse);
+    }
+
+    renderCell(item: taskItem): string {
+        let extraCssClasses = this.opts.extraClass ? this.opts.extraClass(item) : '';
+        
+        return `<div class="cell text-cell ${extraCssClasses}" style="width: ${this.width}">${this.valueProvider(item)}</div>`;
+    }
+    
+    private valueProvider(item: taskItem) {
+        let result = "";
+        
+        if (!!item.nodeTags()) {
+            for (let i = 0; i < item.nodeTags().length; i++) {
```

Comment:
for of instead of for i= 0, whenever possible.

Reply:
for some reason - that did not work here - I tried that first

Reply:
I don't belive that for of doesn't work :)

Reply:
maybe if it will be regular array then it will work

Reply:
we don't want observables here.

Reply:
done